indico_toolkit/indico_wrapper/pred_manipulation.py

```diff
@@ -0,0 +1,115 @@
+from typing import List
+
+
+class ManipulatePreds:
+    """
+    Class to help expand prediction text and indexing via OCR data.
+    """
+
+    def __init__(self, ocr_tokens: List[dict], preds: List[dict]):
+        self.ocr_tokens = ocr_tokens
+        self.predictions = preds
+
+    def expand_predictions(self, pred_start: int, pred_end: int) -> dict:
+        """
+        Expand predictions and boundaries to match that of OCR data.
+
+        Args:
+            pred_start (int): Starting prediction index
+            pred_end (int): Ending prediction index
+        Returns:
+            dict: Returns expanded prediction dictionary
+        """
+        expanded_start, expanded_end = pred_start, pred_end
+        pred_index = None
+
+        # Find index value of bounded prediction
+        for index, pred in enumerate(self.predictions):
+            if pred["start"] == pred_start or pred["end"] == pred_end:
+                pred_index = index
+                break
+
+        if pred_index is None:
+            raise ValueError("No matching prediction found.")
+
+        # Validate current prediction needs expanding
+        original_text = self.predictions[pred_index]["text"]
+        ocr_text_initial = self._get_ocr_text(pred_start, pred_end)
+        if original_text == ocr_text_initial:
+            print("No expansion needed")
+            return self.predictions[pred_index]
+
+        # Use overlapping boundaries and expand text / boundaries to match OCR data
+        for token in self.ocr_tokens:
```

Comment:
you might be able to reuse some token matching functionality from other classes-- and then expand from there

Reply:
e.g. see if you can reuse this class (or if it needs some small tweaks, could include that?) https://github.com/IndicoDataSolutions/Indico-Solutions-Toolkit/blob/main/indico_toolkit/association/extracted_tokens.py